src/os/bluestore/BlueFS.cc

```diff
@@ -3137,6 +3171,9 @@ int BlueFS::_flush_range_F(FileWriter *h, uint64_t offset, uint64_t length)
     h->file->is_dirty = true;
   }
   if (h->file->fnode.size < offset + length) {
+    // here we track inflight flush size. when compact_log is triggered
+    // we correct the fnode's size by excluding its still inflight flush size.
+    h->file->inflight_flush_size += (offset + length) - h->file->fnode.size;
```

Comment:
IMO not every execution path has got a symmetric inflight_flush_size reset.
E.g. _flush_F function (which calls _flush_range_F) might be called from some other functions which lack inflight_flush_size, namely append_try_flush, truncate, fsync

Reply:
thanks for pointing out this issue. for truncate, update to inflight_flush_size is required since _flush_bdev(h) is invoked which guarantees inflight_io is flushed to disk. for append_try_flush, it may not be required since it does not guarantees inflight data are completed after returns. In general, whenever _flush_bdev(h) is called, the inflight_flush_size should be reset.